dvc/remote/ssh/connection.py

```diff
@@ -307,18 +290,14 @@ def cp(self, src, dest):
         self.makedirs(posixpath.dirname(dest))
         self.execute("cp {} {}".format(src, dest))
 
-    @contextmanager
     def open_max_sftp_channels(self):
-        try:
-            channels = []
+        # If there are more than 1 it means we've already opened max amount
```

Comment:
Could you provide a better explanation? Are we only able to open 1 sftp connection over 1 ssh connection? Why did this work before adding connection pool?

Reply:
This the only place we are opening more that 1 sftp connection, so if we have more than 1 then we've already been here and opened max connections.

Before connection pool we were dropping `SSHConnection` after use and open a new one, so no "open max sftp - close all sftp - (reuse ssh conn) - try open sftp - Administratively prohobited" situation.

Reply:
So why do we need _sftp_channels? Why not just `self.sftp`? If we are opening only 1 sftp channel per 1 ssh connection.

Reply:
We are not opening 1 sftp per 1 connection. We are opening as max as we can, but only when we need them. Then we simply not close them and next time we need max channels they are already there.

Reply:
Aaah, right. Got it.